### PR TITLE
[#1025] Combine Split Description

### LINKED ISSUE #1025: [Export] Combine split description
**[Edited on 2022-06-28 to have Prism description contain all DH description separated by double newlines]**
DH records with description split in multiple fields should have those fields combined into a single one at export time.

For example a record originally exported with

```json
"description": "Photograph of an Ola (palm-leaf manuscript) written in Pali around 1760 AD.",
"additional_descriptions": [
        {
          "description": "The binding cord, ivory carved medallion, and lacquered wood covers are modern, made in Ceylon by native artists after ancient models.", 
          "type": {
            "id": "other"
          }
        }, 
        {
          "description": "This ola and another ola were featured on the DigitalHub home page at initial launch in October 2015.", 
          "type": {
            "id": "other"
          }
        }, 
        {
          "description": "[Donated by Dr. Casey Wood, 1934. Photo by James B. Brucker, 2005.] PHOTO Credit line should read: \"Courtesy of the Galter Health Sciences Library Special Collections, Feinberg School of Medicine, Northwestern University, Chicago, Ill.\"", 
          "type": {
            "id": "other"
          }
        }, 
        {
          "description": "About 250 years old, this ola contains information on medical care for conditions for eyes and nose. It prescribes recipes for snuffs for catarrhs, ointments and pastes for eye diseases, and decoctions and powders for nasal and eye diseases.", 
          "type": {
            "id": "abstract"
          }
        }
      ], 
```

should instead be exported as:

```json
"description": "Photograph of an Ola (palm-leaf manuscript) written in Pali around 1760 AD.\n\nThe binding cord, ivory carved medallion, and lacquered wood covers are modern, made in Ceylon by native artists after ancient models.\n\nThis ola and another ola were featured on the DigitalHub home page at initial launch in October 2015.\n\n[Donated by Dr. Casey Wood, 1934. Photo by James B. Brucker, 2005.] PHOTO Credit line should read: \"Courtesy of the Galter Health Sciences Library Special Collections, Feinberg School of Medicine, Northwestern University, Chicago, Ill.\""
"additional_descriptions": [ 
        {
          "description": "About 250 years old, this ola contains information on medical care for conditions for eyes and nose. It prescribes recipes for snuffs for catarrhs, ointments and pastes for eye diseases, and decoctions and powders for nasal and eye diseases.", 
          "type": {
            "id": "abstract"
          }
        }
      ], 
```

### PR DESCRIPTION
**Pull request by Meowcenary**
Join GenericFile#description with double new lines - "\n\n" rather than
taking the first value as the primary description and mapping the rest
to additional descriptions. closes #1025

### CODE REVIEW

**Comment by Meowcenary** [comment]
Simple PR - hopefully I got it right after all my discussion with Guillaume.

File: app/models/invenio_rdm_record_converter.rb [hunk 1]
```diff
@@ -173,9 +173,8 @@ def invenio_metadata
         "creators": @generic_file.creator.map{ |creator| build_creator_contributor_json(creator) },
         "title": @generic_file.title.first,
         "additional_titles": format_additional("title", "alternative-title", @generic_file.title.drop(1)),
-        "description": @generic_file.description.first,
-        "additional_descriptions": format_additional("description", "other", @generic_file.description.drop(1)) + format_additional("description", "acknowledgements", @generic_file.acknowledgments)\
-          + format_additional("description", "abstract", @generic_file.abstract),
+        "description": @generic_file.description.join("\n\n").force_encoding("UTF-8"),
+        "additional_descriptions": format_additional("description", "acknowledgements", @generic_file.acknowledgments) + format_additional("description", "abstract", @generic_file.abstract),
```
Comment (fenekku) [inline app/models/invenio_rdm_record_converter.rb:177]: very minor (i.e. can be merged without) while here to improve readability:

```suggestion
        "additional_descriptions": (
            format_additional(
                "description", "acknowledgements", @generic_file.acknowledgments
            ) +
            format_additional("description", "abstract", @generic_file.abstract)
        ),
```

**Review by fenekku — APPROVED** [review]
That works thanks for bringing up the description merging. (very minor drive-by readability comment - non blocking)

File: app/models/invenio_rdm_record_converter.rb (hunk 1 above)
Comment (Meowcenary) [reply app/models/invenio_rdm_record_converter.rb:177]: I didn't go for exactly this, but something close

File: app/models/invenio_rdm_record_converter.rb [hunk 2]
```diff
@@ -173,9 +173,10 @@ def invenio_metadata
         "creators": @generic_file.creator.map{ |creator| build_creator_contributor_json(creator) },
         "title": @generic_file.title.first,
         "additional_titles": format_additional("title", "alternative-title", @generic_file.title.drop(1)),
-        "description": @generic_file.description.first,
-        "additional_descriptions": format_additional("description", "other", @generic_file.description.drop(1)) + format_additional("description", "acknowledgements", @generic_file.acknowledgments)\
-          + format_additional("description", "abstract", @generic_file.abstract),
+        "description": @generic_file.description.join("\n\n").force_encoding("UTF-8"),
+        "additional_descriptions":
+          format_additional("description", "acknowledgements", @generic_file.acknowledgments) +
```
Comment (Meowcenary) [inline app/models/invenio_rdm_record_converter.rb:178]: Oh, I see - you were considering the readability of the diff AND the code. I'll keep this in mind if we ever get around to implementing a linter for the website.

**Review by carsonicator — APPROVED** [review]
Thanks @Meowcenary

**Pull request merged**